sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/keyvault/common/AzureKeyVaultProperties.java

```diff
@@ -19,6 +19,8 @@ public class AzureKeyVaultProperties extends AbstractAzureHttpConfigurationPrope
      */
     private String endpoint;
 
+    private boolean disableChallengeResourceVerification = false;
```

Comment:
We have a configuration property named `spring.cloud.azure.credential.managed-identity-enabled`, so I am wondering whether we should call this one `challengeResourceVerificationEnabled` and give it a default value `true`.

Spring Boot configuration annotation processor will read the Java Doc on the field to generate the configuration metadata, https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html. So we should add the description to the field as well.

Reply:
Hi,
works for me, thank you! I will update the PR tonight with the renamed property names (both in the PropertySource specific  and the Secret/Certificate specific Properties objects) and will make sure to add JavaDoc to the fields as well!

Reply:
Done, thank you!